### PR TITLE
Replaced deprecated constructors and methods from Thing API

### PR DESCRIPTION
**Pull request by cweitkamp**
- Replaced deprecated constructors and methods from Thing API
- Implement `ChannelGroupTypeProvider` interface in addition to `ChannelTypeProvider` interface

Related to https://github.com/openhab/openhab-core/pull/1414

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java [hunk 1]
```diff
@@ -404,14 +431,17 @@ private void addChannels(KM200ServiceObject serObj, Thing thing, List<Channel> s
...
+                                channelUID, root, CoreItemFactory.NUMBER, null, subKey, subKey, true, false, state,
+                                unitOfMeasure);
                     } else if ("errorString".equals(subKey)) {
-                        state = new StateDescription(null, null, null, "%s", readOnly, null);
+                        state = StateDescriptionFragmentBuilder.create().withPattern("%s").withReadOnly(readOnly)
+                                .build();
                         newChannel = createChannel(new ChannelTypeUID(thing.getUID().getAsString() + ":" + subKey),
-                                channelUID, root, "String", null, "Error message", "Text", true, false, state,
-                                unitOfMeasure);
+                                channelUID, root, CoreItemFactory.NUMBER, null, "Error message", "Text", true, false,
```
Comment (cpmeister) [inline bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java:443]:
```suggestion
                                channelUID, root, CoreItemFactory.STRING, null, "Error message", "Text", true, false,
```

**Comment by cpmeister** [comment]
Manually approved sign.off

File: bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java [hunk 2]
```diff
@@ -324,7 +335,7 @@ private void addChannels(KM200ServiceObject serObj, Thing thing, List<Channel> s
             switch (subKeyType) {
                 case "stringValue":
                     /* Creating an new channel type with capabilities from service */
-                    List<@NonNull StateOption> options = null;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java:327]: This class doesn't have a `@NonNullByDefault` so are you sure that removing `@NonNull` here is safe?

Comment (cweitkamp) [reply bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java:327]: Yes, I am. But it should not be to hard to add it on class level if you prefer.

Comment (cpmeister) [reply bundles/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/handler/KM200ThingHandler.java:327]: Might as well

**Comment by marcelrv** [comment]
LGTM 
for ntp

**Pull request merged**